### PR TITLE
feat(server): Add initial minidump scrubbing

### PR DESCRIPTION
**Pull request by flub**
Implements initial data scrubbing of UTF-8 strings in heap and stack memory streams in minidumps. All sections are marked as `pii = "maybe"`, which makes it effectively disabled for current PII rules.

### CODE REVIEW

File: examples/edit_minidump.rs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+        changed |= pii_processor.scrub_attachment_bytes(&filename, dest, bytes_type);
+    }
+
+    for raw_desc in raw_descriptors {
+        changed |= pii_processor.scrub_attachment_bytes(
+            &filename,
+            data.desc_mut_slice(&raw_desc),
+            AttachmentBytesType::PlainAttachment,
+        );
```
Comment (flub) [inline examples/edit_minidump.rs]: @untitaker this is extremely untested, it just compiles.  The format of these is just a bunch of bytes, which are "unix string" (aka any corrupted encoding possible) separated by NULL bytes.

Comment (untitaker) [reply examples/edit_minidump.rs]: I wonder whether we can get the locale settings out of the minidump to figure out how to decode paths, env, and whatever this is.

Possibly we can find locale settings in the env, so we could try to lossy-decode the env as ascii first, parse out locale, then parse everything (except the memory) as that encoding.

Comment (flub) [reply examples/edit_minidump.rs]: Yes, locale can normally be parsed out of the environ and used to parse out the encoding (my past self seems to think https://hg.sr.ht/~flub/syskit/browse/syskit/_utils.py?rev=tip#L109 is the way to get this, there's probably rust libs though).

Doing this gets back to how to reverse an encoding to modify the minidump though.

**Comment by flub** [comment]
@untitaker @jan-auer would you mind another review mainly looking at:

- how this is integrated in relay.  I'm not sure this is the right place or done in the right way (also, the interface between the two can be done better)

- where the code should live.  I've created a new crate as that was suggested sometime long ago, but it's a bit of a pain to keep the shared deps in sync.  Though in some way having all the processing-feature functionality in a crate seems appealing to me and I don't think that exists yet.  Anyway, if a crate is the way to go i should obviously do this correctly with making it optional and correctly pulled in byt the processing feature.

comments on relay-minidump/src/lib.rs are also welcome, though less necessary currently.  I've left the example untouched for now, but the idea is that it would also use this and becomes a thing cli wrapper only.

**Comment by flub** [comment]
Ok, this is not as finished as I'd like.  Current state:

- Theres a dependency mistake somewhere.  Invoking just `cargo build` fails but `cargo build --workspaces --all-features` does work (which is what I usually do.

- There are some failing integration tests about the minidumps, I think this is an actual bug that should be found

- (minor) The example doesn't use the new module yet

**Review by jan-auer — APPROVED** [review]

**Review by jan-auer — APPROVED** [review]

**Comment by jan-auer** [comment]
@untitaker Can you confirm that this will only scrub minidumps if explicitly configured?

**Review by untitaker — CHANGES_REQUESTED** [review]
This looks fine but we need to figure out how to deploy this without enabling. Right now this will match on selectors `*` and `**`, and the former cannot be solved with selector specificness at the moment. Also if somebody were to have configured a selector named `'UE4Minidump.dmp'`, this would (and should) apply as far as I understand.

I think the sanest way forward would be to have a featureflag inside of the projectconfig that is attached to a featureflag in Sentry.

**Pull request merged**